Tests template

josephShokry opened the pull request.

josephShokry committed:
parsing input using cmd

adding example of how to parse input using cmd

josephShokry committed:
handling json

adding example of handling json data

josephShokry committed:
adding the main file

adding the main file parsing the input (the command and the json) and check the command then execute the proper function

josephShokry committed:
adding more input variables

josephShokry committed:
adding the create function

- adding the create function that makes the directories using the schema
- modifying how to get the schema instead of getting it from the cmd directly now we get it through a text file

MahmoudGoda2003 commented:
nice

MahmoudGoda2003 commented on command.py:
```diff
@@ -0,0 +1,45 @@
+        f=open('jsonschema.json',"r")
+        #print(f.read())
+        data=json.load(f)
+        print(data)
```
you don't need to print data when you create

mohamedabdelsalam replied:
use logging library

MahmoudGoda2003 commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+        f.write(json.dumps(i, indent = 4))
+
+def delete(primary_key):
+    print("Hello from the delete function")
```
delete need to delete the folder not print("Hello from the delete function")

mohamedabdelsalam replied:
"nOT IMPELEMTED YET"

MahmoudGoda2003 commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+def delete(primary_key):
+    print("Hello from the delete function")
+
+def get(primary_key):
```
get and set also don't work

MariaBassem123 commented on command.py:
```diff
@@ -0,0 +1,45 @@
+    parser.add_argument("-sch","--schema",type=str)
+    parser.add_argument("-pk","--primaryKey",type=str)
+    parser.add_argument("-val","--value",type=str)  
+    parser.add_argument("-dp","--data_base",type=str)
```
This should be db stands for data base not dp

mohamedabdelsalam replied:
+1

YahyaAzzam commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+import os
+import json
+def create(schima):
```
What would happen if the directory was already created, will the program crash or should it alert the user without crashing?

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+
+def parseInput():
+    parser=argparse.ArgumentParser()
+    parser.add_argument("-cmd","--command",type=str)
```
What does this args mean ?  Where is the help in each args?
```


parser.add_argument(
--
  | "-s",
  | "--start-time",
  | type=datetime.fromisoformat,
  | help="Start time in ISO format (e.g., '2022-01-04 00:05:23'). Defaults to 7d before end time.",
  | )



```

MariaBassem123 commented on command.py:
```diff
@@ -0,0 +1,45 @@
+        print(data)
+        commands.create(data)
+    elif args.command=="set":
+        commands.set(5,3)
```
I don't understand what are those numbers 5 &3?

mohamedabdelsalam replied:
What is 5 and 3 ?

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+if __name__=='__main__':
+    args=parseInput()
+    data= None
+    if(args.schema is not None):
+        data=json.loads(args.schema)
```
Why do you parse the schema here ? Do you expect schema in each command line input ?

YahyaAzzam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+        f=open('jsonschema.json',"r")
+        #print(f.read())
+        data=json.load(f)
+        print(data)
```
Why would the data be printed for the create function, and would the data be loaded in this file?

MahmoudGoda2003 commented on command.py:
```diff
@@ -0,0 +1,45 @@
+import argparse
+import commands
+
+def parseInput():
```
you should put this function in another file

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+    if(args.schema is not None):
+        data=json.loads(args.schema)
+
+    if args.command=="create":
+        f=open('jsonschema.json',"r")
+        #print(f.read())
+        data=json.load(f)
+        print(data)
+        commands.create(data)
+    elif args.command=="set":
+        commands.set(5,3)
+    elif args.command=="delete":
+        commands.delete(5)
+    elif args.command=="get":
+        commands.get(5)
```
Move this to CommandFactory class. The main should be simple.

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+    if(args.schema is not None):
+        data=json.loads(args.schema)
+
+    if args.command=="create":
```
Do u expect the input case senastive ?

MariaBassem123 commented on jsonschema.json:
```diff
@@ -0,0 +1,12 @@
+{
```
What is the difference between this part and the previous one?

YahyaAzzam reviewed (commented):
Good job but can be better.

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+import json
```
Why did you submit the .pyc. It should be local only in your machine

mohamedabdelsalam commented on csed2025/Students/info.txt:
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "Students",
+    "columns": [
+        "name",
+        "age",
+        "id"
+    ],
+    "primary_key": "id",
+    "Index_keys": [
+        "name"
+    ],
+    "Consistently": "Strong"
```
Wht do you need this ? Is this for test ? If yes, please create a new folder call `test` and add all json and files related to the test

MahmoudGoda2003 commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+    parent_dir=os.getcwd()
+    data_base_name=schima["database_name"]
+    path=os.path.join(parent_dir,data_base_name)
+    os.mkdir(path)
```
this function will return error if the folder Existe

mohamedabdelsalam commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+import os
+import json
+def create(schima):
```
There is two files named `command` and `commands` which is confusing. 
I would suggest `command.py` is the facade design pattern application. and commands should be `command`

mohamedabdelsalam replied:
You should have interface `ICommand` and multiple classes. CreateCommand extends ICommand and so on

MahmoudGoda2003 commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+        os.mkdir(table_path)
+        file_name=os.path.join(table_path,"info.txt")
+        f=open(file_name,'w')
+        f.write(json.dumps(i, indent = 4))
```
you didn't cover if you want to add new table so you should create it, or if some one delete any folder by mistake this will generate error in your code

mohamedabdelsalam commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+import json
+def create(schima):
+    parent_dir=os.getcwd()
+    data_base_name=schima["database_name"]
```
Why don't you have an enum called SCHEMA_NAME. instead of hardcode here ?

mohamedabdelsalam replied:
class SchemaKeys 

final static String TABLES = "Tables"

mohamedabdelsalam commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+    data_base_name=schima["database_name"]
+    path=os.path.join(parent_dir,data_base_name)
+    os.mkdir(path)
+    for i in schima["Tables"]:
```
`for table in schema["Tables"]`
make the variable meaningful.

bangkook commented on command.py:
```diff
@@ -0,0 +1,45 @@
+
+def parseInput():
+    parser=argparse.ArgumentParser()
+    parser.add_argument("-cmd","--command",type=str)
```
I think command is optional this way.

mohamedabdelsalam replied:
+1

mohamedabdelsalam commented on commands.py:
```diff
@@ -0,0 +1,22 @@
+    os.mkdir(path)
+    for i in schima["Tables"]:
+        table_path=os.path.join(path,i["name"])
+        os.mkdir(table_path)
+        file_name=os.path.join(table_path,"info.txt")
+        f=open(file_name,'w')
```
Did you. check if the dir exists? 
Why don't you have a private method call create folder instead of repeating this?

mohamedabdelsalam commented on tempCodeRunnerFile.py:
```diff
@@ -0,0 +1 @@
+fname
```
Why this ?

mohamedabdelsalam commented on parse.py:
```diff
@@ -0,0 +1,11 @@
+import argparse
+import os
+import string
+from tokenize import String
+
+parser = argparse.ArgumentParser(description="testing the parsing in python")
+parser.add_argument('words',type=str,help="to parse the input and print it")
+args=parser.parse_args()
+
+if __name__=='__main__':
+    print (args.words)
```
What is this file ?

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+if __name__=='__main__':
+    args=parseInput()
+    data= None
+    if(args.schema is not None):
```
space between if (

mohamedabdelsalam commented on command.py:
```diff
@@ -0,0 +1,45 @@
+        data=json.loads(args.schema)
+
+    if args.command=="create":
+        f=open('jsonschema.json',"r")
```
What is this ?

josephShokry committed:
create funciton

completing the create function, making the file of the parser and cleaning the code

josephShokry committed:
handling errors of invalid schema file

if the user parsed a not existent file of not parsing the schema file at all

josephShokry committed:
Delete __pycache__ directory

josephShokry committed:
adding command factory and help in parser

i merged the code of hussain about the command factory with the parser the main code and adder the help attribute of the parsers

josephShokry committed:
.

josephShokry committed:
Merge branch 'tests-template' of https://github.com/josephShokry/SimpleFSDB into tests-template

mohamedabdelsalam commented on Parser.py:
```diff
@@ -0,0 +1,21 @@
+import argparse
```
Please remove __pycache__ directory

mohamedabdelsalam commented on Parser.py:
```diff
@@ -0,0 +1,21 @@
+parent_path=os.getcwd()
+def parseInput():
+    parser=argparse.ArgumentParser()
+    parser.add_argument("-cmd","--command",type=str,help="the quiry command")
```
leave space after `,` in all lines

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+        elif command_type == "set":
+            return set()
+
+class create(Icommand):
```
I recommend every command class in a file. 
u will have create_command.py
get_command.py. 

also class name should be CreateCommand. 


class name should start with capital

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+    def excute():
+        """the interface Icommand"""
+
+class commandfactory:
```
start with Capital letter

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+
+class create(Icommand):
+    def excute(self,args):
+        f=open(args.schema,"r")
```
variable name convension. don't use variable as `f`

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+class create(Icommand):
+    def excute(self,args):
+        f=open(args.schema,"r")
+        schima=json.load(f)
```
schema

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+        f=open(args.schema,"r")
+        schima=json.load(f)
+        parent_dir=os.getcwd() #get the current dir of the project
+        data_base_name=schima["database_name"]
```
create SCHEMA_KEy class

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+        parent_dir=os.getcwd() #get the current dir of the project
+        data_base_name=schima["database_name"]
+        path=os.path.join(parent_dir,data_base_name)
+        if not os.path.isdir(path): #to check if the DB directory is created before or not
+            os.mkdir(path)
```
mkdir takes boolean, instead those if condition

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+        path=os.path.join(parent_dir,data_base_name)
+        if not os.path.isdir(path): #to check if the DB directory is created before or not
+            os.mkdir(path)
+        for i in schima["Tables"]:
```
for table in schema["tables"]

mohamedabdelsalam commented on commands_function.py:
```diff
@@ -0,0 +1,49 @@
+            os.mkdir(path)
+        for i in schima["Tables"]:
+            table_path=os.path.join(path,i["name"])
+            if not os.path.isdir(table_path):
+                os.mkdir(table_path)
```
same above

josephShokry committed:
adding the keys class

adding the keys class to simplify the coding

josephShokry committed:
adding test cases

adding test cases of create function to check if all the tables and DB has been created or not

hussainmansour committed:
merging between test-template and AvoidingError

josephShokry committed:
Delete __pycache__ directory
